src/login/logind-dbus.c

```diff
@@ -3256,6 +3115,15 @@ static int method_set_wall_message(
         if (r < 0)
                 return r;
 
+        /* sysvinit has a 252 (256-(strlen(" \r\n")+1)) character
+         * limit for the wall message. There is no real technical
+         * need for that but doesn't make sense to store arbitrary
+         * armounts either.
+         * https://git.savannah.nongnu.org/cgit/sysvinit.git/tree/src/shutdown.c#n72)
+        */
+        if (strlen(wall_message) > 252)
+                return -EMSGSIZE;
```

Comment:
this is kinda a compat breakage: so far we accepted that without complaining, suddenly if someone passes a long string we'll just fail.

it probably makes sense to enforce a limit here, but maybe larger? 252 sounds awfully short. maybe enforce 4k or so? and then document in the man page that it would be wise not to pass more than 252 because this is a sysvinit interface?

Reply:
I think we should pick a limit that is unlikely to interfere with what people actually intend to do. i.e. so if we assume people want to write a word we'll accept a sentence and if they want a sentence we'll accept a paragraph, and if they want a paragraph we'll take a novel. thus, it doesn't really matter if 4k is a novel, what matters if 252 is enough for the usual sentences in the world's many languages. And I doubt it would be, given UTF8, and stuff. Thus I'd suggest 4K precisely for the reason that it's unlikely people will need that much. hope that thinking makes sense?

Reply:
sure